src/xpcc/architecture/platform/driver/adc/stm32f0/adc.hpp.in

```diff
@@ -0,0 +1,260 @@
+// coding: utf-8
+/* Copyright (c) 2013, Roboterclub Aachen e.V.
+* All Rights Reserved.
+*
+* The file is part of the xpcc library and is released under the 3-clause BSD
+* license. See the file `LICENSE` for the full license governing this code.
+*/
+// ----------------------------------------------------------------------------
+
+#ifndef XPCC_STM32F0_ADC_HPP
+#define XPCC_STM32F0_ADC_HPP
+
+#include <stdint.h>
+#include "../../../type_ids.hpp"
+#include "../../../device.hpp"
+#include <xpcc/architecture/interface/register.hpp>
+
+namespace xpcc
+{
+namespace stm32
+{
+/**
+ * Analog/Digital-Converter module (ADC1).
+ *
+ * The 12-bit ADC is a successive approximation analog-to-digital
+ * converter. It has up to 19 multiplexed channels allowing it measure
+ * signals from 16 external and three internal sources.
+ * The result of the ADC is stored in a left-aligned or right-aligned
+ * 16-bit data register.
+ *
+ * This API is designed for the internal ADCs of STM32F0x1/STM32F0x2/STM32F0x8
+ *
+ * \author	Álan Crístoffer
+ * \ingroup	stm32
+ */
+class Adc
+{
+public:
+%% for channel in range(0,16)
```

Comment:
Not all STM32 F0 have 16 channels. Some only have 10. [STM32F038x6](http://www.st.com/content/ccc/resource/technical/document/datasheet/5f/5d/72/7b/c3/2d/46/54/DM00110868.pdf/files/DM00110868.pdf/jcr:content/translations/en.DM00110868.pdf), [STM32F042x4](http://www.st.com/content/ccc/resource/technical/document/datasheet/5f/5d/72/7b/c3/2d/46/54/DM00110868.pdf/files/DM00110868.pdf/jcr:content/translations/en.DM00110868.pdf), [STM32F048C6](http://www.st.com/content/ccc/resource/technical/document/datasheet/c4/8f/82/19/35/83/43/45/DM00109264.pdf/files/DM00109264.pdf/jcr:content/translations/en.DM00109264.pdf)

Reply:
The code I took the template from had a channels variable defined, but I could not track it down. Is it possible to take the number of channels from the device configuration files?

Reply:
The `channels` variable was manually added. The device XMLs contain the information, but it is not accessible from within the jinja template engine.

Reply:
Leave it like this for now, but leave a TODO comment about this. xpcc doesn't have the right tools to conceptually fix this data problem anyways.